src/app/twin/page.tsx

```diff
@@ -0,0 +1,169 @@
+'use client'
+
+import {toast} from "sonner";
+import {ColumnDef, PaginationState} from "@tanstack/table-core";
+import {DataTableHandle} from "@/components/base/data-table/data-table";
+import {useContext, useRef, useState} from "react";
+import {TwinBase} from "@/lib/api/api-types";
+import {ApiContext} from "@/lib/api/api";
+import {ShortGuidWithCopy} from "@/components/base/short-guid";
+import {CrudDataTable, FiltersState} from "@/components/base/data-table/crud-data-table";
+import {useRouter} from "next/navigation";
+import {buildFilters, FilterFields, FILTERS} from "@/entities/twin"
+import {useTwinList} from "@/shared/hooks/useTwinList";
+import {components} from "@/lib/api/generated/schema";
+
+type FetchDataResponse = {
+    data: TwinBase[],
+    pageCount: number
+};
+
+type TwinList = components['schemas']['TwinSearchRsV2']['twinList']
+
+const columns: ColumnDef<TwinBase>[] = [
+    {
+        accessorKey: "id",
+        header: "ID",
+        cell: (data) => <ShortGuidWithCopy value={data.row.original.id}/>
+    },
+    {
+        accessorKey: "createdAt",
+        header: "Created at",
+    },
+    {
+        accessorKey: "authorUserId",
+        header: "Author User Id",
+        cell: (data) => <ShortGuidWithCopy value={data.row.original.authorUserId}/>
+    },
+    {
+        accessorKey: "name",
+        header: "Name",
+    },
+    {
+        accessorKey: "assignerUserId",
+        header: "Assigner User Id",
+        cell: (data) => <ShortGuidWithCopy value={data.row.original.assignerUserId}/>
+    },
+    {
+        accessorKey: "twinClassId",
+        header: "Twin Class Id",
+        cell: (data) => <ShortGuidWithCopy value={data.row.original.twinClassId}/>
+    },
+]
+
+export default function Twin() {
+    const [classDialogOpen, setClassDialogOpen] = useState(false)
+
+    const api = useContext(ApiContext)
+    const router = useRouter()
+    const tableRef = useRef<DataTableHandle>(null);
+    const {data} = useTwinList<{ twinList: TwinList }>(api);
+    const twinIds: string[] = data?.twinList
+        ? data.twinList?.map(i => i.id ?? 'N/A')
+        : []
+    const twinIdOptions = twinIds.map(id => ({
+        id,
+        name: id.length > 12 ? `${id.slice(0, 12)}...` : id
+    }))
```

Comment:
Are `twinIdOptions` used anywhere on the page? If not, please remove it, as well as `useTwinList.ts` file if it is not used